src/OrchardCore.Modules/Orchard.Autoroute/Drivers/AutoroutePartDisplay.cs

```diff
@@ -61,15 +73,36 @@ public override IDisplayResult Edit(AutoroutePart autoroutePart)
         public override async Task<IDisplayResult> UpdateAsync(AutoroutePart model, IUpdateModel updater)
         {
             await updater.TryUpdateModelAsync(model, Prefix, t => t.Path);
-            
+
             var httpContext = _httpContextAccessor.HttpContext;
-            
             if (httpContext != null && await _authorizationService.AuthorizeAsync(httpContext.User, Permissions.SetHomepage))
             {
                 await updater.TryUpdateModelAsync(model, Prefix, t => t.SetHomepage);
             }
 
+            await ValidateAsync(model, (key, message) => updater.ModelState.AddModelError(Prefix, key, message));
+
             return Edit(model);
         }
+
+        private async Task ValidateAsync(AutoroutePart autoroute, Action<string, string> reportError)
+        {
+            var invalidCharacters = ":?#[]@!$&'()*+,.;=<>\\|% ".ToCharArray();
+            if (autoroute.Path.IndexOfAny(invalidCharacters) > -1)
+            {
+                var invalidCharactersForMessage = string.Join(", ", invalidCharacters.Select(c => $"\"{c}\""));
+                reportError(nameof(autoroute.Path), T["Please do not use any of the following characters in your permalink: {0}. No spaces are allowed (please use dashes or underscores instead).", invalidCharactersForMessage]);
```

Comment:
You should split it in two tests, one for special chars, and one for spaces. And remove the space from the special chars list.